integration-tests/utils/analysis-utils.js

```diff
@@ -0,0 +1,7 @@
+const findButtonInCloudEnvDetailsDialog = (page, ariaLabelButtonText) => {
+  return page.waitForXPath(`//*[@role="dialog" and @aria-hidden="false"]//*[@role="button" and contains(@aria-label,"${ariaLabelButtonText}")]`,
+    { visible: true }
+  )
+}
```

Comment:
1. Does this function have anything specific to CloudEnvironmentModal? If not, I'd suggest moving it to `integration-utils.js`?
2. I also see that `getAnimatedDrawer` in `integration-utils.js` does a similar search and there doesn't seem to be anything specific in that function about an animated drawer so I wonder if we could refactor `getAnimatedDrawer` to work for its current use cases as well as the new case you're adding  🤔

Reply:
Hmm, it could be useful for other tests but I don't know any right now. 
Can you tell me:
1) What other tests beside analysis tests can use this finder function?
2) How many other UI elements (in other UI pages) where you should find elements by `aria-label` (not by visible text) in a dialog?

I'm happy to move it to `integration-utils` file if it will be needed by non-analysis tests. But we shouldn't do it if we thought it would be needed but didn't end up being needed at all. 🙂

Reply:
1. I think I hadn't asked my question clearly. Can we rename the helper as something like `findButtonByAriaLabel` so it's clear that it could be used in _any_ modal as long as the purpose is matching some aria-label text?
2. I agree with you about not prematurely factoring code out. For the same reason, the helper could be placed in the only file it's used from (`analysis-context-bar.js`) but I don't feel strongly.

Reply:
Rename function to `findButtonInDialogByAriaLabel` and moved it to `integration-utils.js`.